hash_practice/exercises.py

```diff
@@ -2,18 +2,41 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: n^2logn
+        Space Complexity: n
     """
-    pass
+    freq_map = {}
+    for word in strings:  # n
+        key = list(word) # n
+        key.sort()  # logn
+        key = "".join(key)  # n
+        if key in freq_map:  # n
+            freq_map[key].append(word)
+        else:
+            freq_map[key] = [word]
+    return [val for val in freq_map.values()]
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: nlogn
+        Space Complexity: n
     """
-    pass
+    freq_dict = {}
+    for num in nums: # n
+        freq_dict[num]= freq_dict.get(num, 0)+1
+    
+    freq_target = [freq for freq in freq_dict.values()] # n
+    freq_target.sort() # logn

```

Comment:
Sorts like this usually O(n*log(n)) not just O(log(n)). Basically while multiple structures can handle inserts in O(log(n)) time, any algorithm still needs to go through all the entries in the list, so n times for O(n*log(n)).